R/shared-methods.R

```diff
@@ -509,6 +513,53 @@ DataClass <- R6::R6Class(
         source_data_cols = paste(unlist(self$source_data_cols), collapse = ", ")
       )
       return(sum_df)
+    },
+
+    #' @description Run tests on a country class instance. Calling `test()` on a
+    #' class instance runs tests with the settings in use. For example, if you
+    #' set `level = "1"` and `localise = FALSE` the tests will be ran on level 1
+    #' data which is not localised. Rather than downloading data for a test
+    #' users can provide a path to a snapshot file of data to test instead.
+    #' Tests are ran on a clone of the class. This method calls generic tests
+    #' for all country class objects. It also calls country specific tests
+    #' which can be defined in an individual country class method called
+    #' `specific_tests()`.
+    #' @param download logical. To download the data (TRUE) or use a snapshot
+    #' (FALSE). Defaults to FALSE.
+    #' @param snapshot_path character_array the path to save the downloaded data
+    #' or read from. Must be a path to an rds file.
+    #' @param test_all logical. Run tests with all settings (TRUE) or with those
```

Comment:
sorry so fussy. This is inside a function call test, does it need to be called test_all or just all?